packages/fxa-content-server/app/scripts/views/subscriptions_product_redirect.js

```diff
@@ -15,30 +15,45 @@ class SubscriptionsProductRedirectView extends FormView {
   initialize(options) {
     this._currentPage = options.currentPage;
     this._productId = this._currentPage.split('/').pop();
-    this._queryParams = Url.searchParams(this.window.location.search);
     this.relier.set('subscriptionProductId', this._productId);
 
     this._subscriptionsConfig = {};
     if (options && options.config && options.config.subscriptions) {
       this._subscriptionsConfig = options.config.subscriptions;
     }
 
-    this.mustAuth =
-      !!this._queryParams.signin ||
-      !this._subscriptionsConfig.allowUnauthenticatedRedirects;
-
     // Flow events need to be initialized before the navigation
     // so the flow_id and flow_begin_time are propagated
     this.initializeFlowEvents();
   }
 
+  render() {
```

Comment:
I think you might be able to do this in the `beforeRender`?

Reply:
Sure if it returns a promise that is awaited; that's a good point. I'll move it.

Reply:
The Base view's `render` calls `beforeRender` _after_ it calls `checkAuthorization`, but `checkAuthorization` need the `mustAuth` value, so that's why `mustAuth` needs to be determined before all of that happens.

Reply:
So `render` [gets executed before `beforeRender`](https://github.com/mozilla/fxa/blob/32c09728bfc2f5ad8208744c76f406045dfbbd8c/packages/fxa-content-server/app/scripts/views/base.js#L264-L274)? How does that make any sense?

Reply:
I think that given the name of these functions, we _naturally_ imagine that there's some lifecycle management code calling `beforeRender`, `render`, and then `afterRender`, in that order.  But that's not the case.  The only framework (Backbone.js) function is `render`; that's gets call because it's the rendering function.

Reply:
You are correct though, Barry.

Reply:
I guess I'm thinking this code looks like it could be simplified a little. `navigateToPaymentServer` also calls `isSignedIn`. Maybe we could skip the promises here and just check if `this.getSignedInAccount();` returns an object, either way this is just a nit an nothing to block on :)

Reply:
I don't think we can correctly set `mustAuth` without checking `isSignedIn`?

But yeah, maybe we can passed the signed in state so we don't need to call `isSignedIn` twice.  (And agreed that it's not blocking.)

Reply:
Thanks Vijay for your thoughts here.

I'm not sure I understand well enough the difference between the `getSignedInAccount` and `account.isSignedIn` methods. Is `getSignedInAccount` sufficient to say that an account is signed in if an object is returned? If so, what is the difference between that and `account.isSignedIn()` (other than their return values)?

I know this promise stuff is a little confusing (honestly more than a little). I'm inclined to move forward here with what we have, as I know some folks are anxious to get things merged, even if they aren't perfect. But I'm happy to chat with you more about this and potentially submit a follow-up later if that works for you.